src/audio/drc/drc.c

```diff
@@ -140,78 +143,70 @@ static int drc_setup(struct drc_comp_data *cd, uint16_t channels, uint32_t rate)
  * End of DRC setup code. Next the standard component methods.
  */
 
-static struct comp_dev *drc_new(const struct comp_driver *drv,
-				const struct comp_ipc_config *config,
-				const void *spec)
+static int drc_init(struct processing_module *mod)
 {
-	struct comp_dev *dev = NULL;
+	struct module_data *md = &mod->priv;
+	struct comp_dev *dev = mod->dev;
+	struct module_config *cfg = &md->cfg;
 	struct drc_comp_data *cd = NULL;
```

Comment:
init not necessary

Reply:
It was in original. I think it is to ensure a free is done for null pointer.